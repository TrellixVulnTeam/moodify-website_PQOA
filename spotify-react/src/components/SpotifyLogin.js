import React from "react";

export default function SpotifyLogin() {
    const CLIENT_ID = "ed61ad2eb1ac48f38a5971328cec9f01"
    const REDIRECT_URI = "http://localhost:8080"
    const AUTH_ENDPOINT = "https://accounts.spotify.com/authorize"
    const RESPONSE_TYPE = "token"
    const SCOPE = "playlist-modify-private user-read-recently-played"

    return (
        <div data-testid="spotify-login">
            <button>
                <a className="rounded-full spotify-login-button py-3 px-6"
                   href={`${AUTH_ENDPOINT}?client_id=${CLIENT_ID}&redirect_uri=${REDIRECT_URI}&response_type=${RESPONSE_TYPE}&scope=${SCOPE}`}>
                    <i className="fa fa-spotify"></i>   Login to Spotify
                </a>
            </button>
        </div>


    )
}

