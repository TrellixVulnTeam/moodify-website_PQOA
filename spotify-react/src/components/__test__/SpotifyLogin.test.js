import React from "react";
import ReactDOM from 'react-dom'
import SpotifyLogin from '../SpotifyLogin'

import {cleanup, getByTestId, render} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<SpotifyLogin/>, div)
})

afterEach(cleanup);

it("renders empty recent tracks correctly", () => {
    const {getByTestId} = render(<SpotifyLogin/>)
    expect(getByTestId("spotify-login")).toBeDefined()
})

it("matches snapshot", () => {
    const tree = renderer.create(<SpotifyLogin/>).toJSON();
    expect(tree).toMatchSnapshot();
})


