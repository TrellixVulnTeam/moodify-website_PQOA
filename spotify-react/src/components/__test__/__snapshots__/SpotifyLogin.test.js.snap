// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="spotify-login"
>
  <button>
    <a
      className="rounded-full spotify-login-button py-3 px-6"
      href="https://accounts.spotify.com/authorize?client_id=ed61ad2eb1ac48f38a5971328cec9f01&redirect_uri=http://localhost:8080&response_type=token&scope=playlist-modify-private user-read-recently-played"
    >
      <i
        className="fa fa-spotify"
      />
         Login to Spotify
    </a>
  </button>
</div>
`;
